app/models/work_paper.rb

```diff
@@ -275,6 +295,22 @@ def unzip_if_necesary
     end
   end
 
+   def sanitized_previous_code
+     pre_code = @previous_code.sanitized_for_filename
+  end
+
+  def check_previous_code file_name
```

Comment:
```suggestion
  def start_with_code? filename
```

Y que verifique por el viejo y el nuevo.